hive-metastore/src/main/java/org/apache/iceberg/hive/CachedClientPool.java

```diff
@@ -33,16 +33,18 @@
 
 public class CachedClientPool implements ClientPool<IMetaStoreClient, TException> {
 
+  @VisibleForTesting static final String CATALOG_DEFAULT = "metastore.catalog.default";
   private static Cache<String, HiveClientPool> clientPoolCache;
 
   private final Configuration conf;
-  private final String metastoreUri;
+  private final String clientPoolKey;
   private final int clientPoolSize;
   private final long evictionInterval;
 
   CachedClientPool(Configuration conf, Map<String, String> properties) {
     this.conf = conf;
-    this.metastoreUri = conf.get(HiveConf.ConfVars.METASTOREURIS.varname, "");
+    this.clientPoolKey =
+        conf.get(HiveConf.ConfVars.METASTOREURIS.varname, "") + conf.get(CATALOG_DEFAULT, "");
```

Comment:
I would prefer it because what about my two metastores

`host1:9083 with default catalog ""`
And my metastore
`host1:90 with default catalog "83"`

On that note are you allowed to do a metastore uri without a port?